hasher.go

```diff
@@ -94,7 +94,7 @@ func (h *Hasher) Reset() {
 	h.hash.Reset()
 }
 
-func (h *Hasher) appendBytes32(b []byte) {
+func (h *Hasher) AppendBytes32(b []byte) {
```

Comment:
I made `AppendBytes32` an exported method so that it could be used directly by generated code that needs to pad non-aligned ByteLists, like the `ExecutionPayload.Transactions` list elements.